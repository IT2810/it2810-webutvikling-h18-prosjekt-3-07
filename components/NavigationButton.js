import React from 'react';
import {StyleSheet, Dimensions, TouchableOpacity, Text} from 'react-native';

const screenWidth = Dimensions.get('window').width;

//Button component used on HomeScreen
class NavigationButton extends React.Component {
    render() {
        return (
            <TouchableOpacity 
            style={styles.button}
            onPress = {this.props.onPress} > 
            <Text style={styles.buttonText}>{this.props.title}</Text>
          </TouchableOpacity>
        )
    }
}

const styles = StyleSheet.create({
    button: {
        width: 0.8*screenWidth, 
        height: 40,
        backgroundColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 10,
        borderColor: '#8492A6',
        borderWidth: 2,
        marginBottom: 10
        
    },
    buttonText: {
        fontSize: 19,
        color: 'black'
    },
})

export default NavigationButton;