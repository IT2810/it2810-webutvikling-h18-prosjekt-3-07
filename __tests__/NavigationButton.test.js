import React from "react";
import renderer from "react-test-renderer";
import NavigationButton from "../components/NavigationButton";

it("renders correctly (navigationButton", () => {
  const tree = renderer.create(<NavigationButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
