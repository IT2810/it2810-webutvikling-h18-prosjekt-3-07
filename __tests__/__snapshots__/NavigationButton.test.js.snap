// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (navigationButton 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "#8492A6",
      "borderRadius": 10,
      "borderWidth": 2,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "width": 600,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 19,
      }
    }
  />
</View>
`;
